modules/internal/ChapelArray.chpl

```diff
@@ -1156,6 +1156,9 @@ module ChapelArray {
     pragma "no doc"
     pragma "no copy return"
     proc buildArray(type eltType) {
+      if eltType == void {
+        compilerError("array element type cannot be void");
+      }
```

Comment:
Perhaps another semantic issue: Should arrays of void simply become void variables themselves?  (or arrays of void variables?)